test/dotnetup.Tests/LibraryTests.cs

```diff
@@ -60,4 +63,169 @@ public void TestGetSupportedChannels()
         channels.Should().NotContain("7.0.1xx");
 
     }
+
+    [Fact]
+    public void MuxerIsUpdated_WhenInstallingNewerSdk()
+    {
+        var releaseInfoProvider = InstallerFactory.CreateReleaseInfoProvider();
+        var installer = InstallerFactory.CreateInstaller(new NullProgressTarget());
+
+        using var testEnv = DotnetupTestUtilities.CreateTestEnvironment();
+
+        Log.WriteLine($"Installing to path: {testEnv.InstallPath}");
+
+        // Install .NET SDK 9.0 first
+        var sdk9Version = releaseInfoProvider.GetLatestVersion(InstallComponent.SDK, "9.0");
+        Log.WriteLine($"Installing .NET SDK 9.0: {sdk9Version}");
+        installer.Install(
+            new DotnetInstallRoot(testEnv.InstallPath, InstallerUtilities.GetDefaultInstallArchitecture()),
+            InstallComponent.SDK,
+            sdk9Version!);
+
+        var muxerPath = Path.Combine(testEnv.InstallPath, DotnetupUtilities.GetDotnetExeName());
+        File.Exists(muxerPath).Should().BeTrue("muxer should exist after SDK 9.0 installation");
+
+        var muxerHashAfterSdk9 = GetFileHash(muxerPath);
+        var muxerSizeAfterSdk9 = new FileInfo(muxerPath).Length;
+        Log.WriteLine($"Muxer after SDK 9.0 install - Size: {muxerSizeAfterSdk9}, Hash: {muxerHashAfterSdk9}");
+
+        Version? fileVersionAfterSdk9 = null;
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            fileVersionAfterSdk9 = GetMuxerFileVersion(muxerPath);
+            Log.WriteLine($"Muxer FileVersion after SDK 9.0 install: {fileVersionAfterSdk9}");
+        }
+
+        // Install .NET SDK 10.0 second
+        var sdk10Version = releaseInfoProvider.GetLatestVersion(InstallComponent.SDK, "10.0");
+        Log.WriteLine($"Installing .NET SDK 10.0: {sdk10Version}");
+        installer.Install(
+            new DotnetInstallRoot(testEnv.InstallPath, InstallerUtilities.GetDefaultInstallArchitecture()),
+            InstallComponent.SDK,
+            sdk10Version!);
+
+        var muxerHashAfterSdk10 = GetFileHash(muxerPath);
+        var muxerSizeAfterSdk10 = new FileInfo(muxerPath).Length;
+        Log.WriteLine($"Muxer after SDK 10.0 install - Size: {muxerSizeAfterSdk10}, Hash: {muxerHashAfterSdk10}");
+
+        Version? fileVersionAfterSdk10 = null;
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            fileVersionAfterSdk10 = GetMuxerFileVersion(muxerPath);
+            Log.WriteLine($"Muxer FileVersion after SDK 10.0 install: {fileVersionAfterSdk10}");
+        }
+
+        // Verify muxer was updated (file changed)
+        muxerHashAfterSdk10.Should().NotBe(muxerHashAfterSdk9, "muxer file should be updated when installing newer SDK");
+
+        // On Windows, also verify FileVersion was upgraded
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && fileVersionAfterSdk9 != null && fileVersionAfterSdk10 != null)
```

Comment:
Let's assert that on Windows the file version before and after the upgrade should not be null, instead of skipping the test if they are null.

Same thing for the other test.